Implement the new design of the Register second passphrase page - Closes #2136

slaweet opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

### What issue have I solved?
<!--- Complementary description if needed -->
#2136

### How have I implemented/fixed it?
<!--- Describe your technical implementation -->
- [x] Refactored  first passphrase backup step to have to common behaviour in [passphraseBackup](https://github.com/LiskHQ/lisk-hub/tree/70a4e4a0cb647fa8864dd4d05a30c8c9ad4ba422/src/components/passphraseBackup)
- [x] Implemented the backup passphrase step (using `PassphraseBackup` component) https://projects.invisionapp.com/d/main#/console/17570736/366378003/inspect
- [x] Implemented summary step  (using `TransactionSummary` Component) https://projects.invisionapp.com/d/main#/console/17570736/366378003/inspect Note that it is without box, as all other transaction types don't have the box yet and should be updated to use the BoxV2 in the summary step all at once in one ticket.
- [x] Implemented transaction result step
- [x] Implemented error toast insufficient balance (same as in voting)
- [x] Updated e2e tests
- [x] Deleted v1 1st and 2nd passphrase components as no-longer-needed
- [x] Deleted secondPassphrase redux as no-longer-needed


### How has this been tested?
<!--- Please describe how you tested your changes. -->
- Go to settings 
- try to register 2nd passphrase
- follow the steps

### Review checklist
- The PR follows our [Test guide](/LiskHQ/lisk-hub/blob/development/docs/TEST_GUIDE.md)
- The PR follows our [CSS guide](/LiskHQ/lisk-hub/blob/development/docs/CSS_GUIDE.md)

osvaldovega commented on src/components/passphraseBackup/index.js:
```diff
@@ -0,0 +1,136 @@
+                <QRCode value={account.passphrase} />
+              </div>
+              <div className={styles.downloadLisk}>
+                <img src={svgIcons.fileOutline} />
```
If I am not wrong I think we can use here the component that you use for manage the svgs

osvaldovega commented on src/components/registerV2/backupPassphrase.js:
```diff
@@ -1,65 +1,16 @@
-import moment from 'moment';
+import React from 'react';
 import grid from 'flexboxgrid/dist/flexboxgrid.css';
+
```
I dont know if this new line was intended or not, but if not maybe you can fix it

slaweet replied:
It was inserted by [import-js](https://github.com/Galooshi/import-js) which I use since a few weeks to automatically generate imports for me. I can recommend the tool for you and @massao too, it has plugins for various editors.

 I changed config not to insert the empty line between internal and external imports https://github.com/LiskHQ/lisk-hub/pull/2142/commits/7fc14b1e61daf3616c3450d032c12e4ff68bd60e

osvaldovega commented on src/components/registerV2/backupPassphrase.js:
```diff
@@ -1,65 +1,16 @@
 import registerStyles from './registerV2.css';
-import styles from './backupPassphrase.css';
 
 class BackupPassphrase extends React.Component {
```
This component now looks like just a presentational component so maybe instead of a class you can use a stateless component function and then just make the destructuring in the parameters and not use the class.
I think looks better what do you think?

osvaldovega commented on src/components/secondPassphrase/secondPassphrase.js:
```diff
@@ -1,19 +1,31 @@
+
+    // istanbul ignore next
+    const crypotObj = window.crypto || window.msCrypto;
+    this.secondPassphrase = generatePassphrase({
```
I think this function will looks better in a utils file and then just called in here

slaweet replied:
Good catch. Fixed.

osvaldovega commented on src/actions/account.js:
```diff
@@ -75,27 +74,33 @@ export const passphraseUsed = data => ({
+  secondPassphrase, account, passphrase, callback,
+}) =>
   (dispatch, getState) => {
+    if (account.balance < Fees.setSecondPassphrase) {
```
I think this validation should be done outside this thunk and be called before this function.
This function should be only on charge of register the second pass, but I think will be better to avoid that one function manage more than one responsibility. What do you think?

slaweet replied:
I checked with Dominic and implemented checking the balance already on settings page.

slaweet commented:
Renamed https://github.com/LiskHQ/lisk-hub/pull/2142/commits/c4848aaa449cf87f6045ad8c226e3d392e647385 @yasharAyari

slaweet commented:
Please check again @Efefefef 
All issues should be fixed except:
- Button widths
<img width="814" alt="Screenshot 2019-06-21 at 10 59 11" src="https://user-images.githubusercontent.com/1254342/59911369-06737f00-9414-11e9-9e84-688e4df4d371.png">
- Box in summary page - this one will be updated in all transaction types together to keep consistency, send and vote summary still have the design without box.

slaweet commented:
Fixed all except this one:

> 🐛 Wrong size
> ![image](https://user-images.githubusercontent.com/6704159/59912210-db8a2a80-9415-11e9-982c-c96aaff12659.png)

Invissison says 15px and it is 15px. But I noticed the color wasn't correct, so I fixed that.

Efefefef commented:
Hmm height is 18 in Invision, and 20 in app

Looks like same problem here plus text should be vertically aligned to center 
![image](https://user-images.githubusercontent.com/6704159/59919979-498c1d00-9429-11e9-956b-a42c87d19e95.png)
Same here
![image](https://user-images.githubusercontent.com/6704159/59920013-6cb6cc80-9429-11e9-8e44-96c846d426ec.png)

slaweet commented:
- Fixed vertical alignment
- the `OR` line is `rgba(0,0,0,0.15);` in design. Now I used it, but we don't have such color in the styleguide https://projects.invisionapp.com/d/main#/console/17570736/365298851/inspect so should be updated in design.
